src/snitch_append.cpp

```diff
@@ -1,65 +1,74 @@
+    if (err != std::errc{}) {
+        // Not enough space, try into a temporary string that *should* be big enough,
+        // and copy whatever we can. 32 characters is enough for all integers and floating
+        // point values encoded on 64 bit or less.
+        small_string<32> fallback;
+        auto [end2, err2] = std::to_chars(
+            fallback.end(), fallback.begin() + fallback.capacity(), value, fmt, precision);
+        if (err2 != std::errc{}) {
+            return false;
+        }
+        fallback.grow(end2 - fallback.begin());
+        return append(ss, fallback);
     }
-}
 
-template<typename T>
-bool append_fmt(small_string_span ss, T value) noexcept {
-    if (ss.available() <= 1) {
-        // snprintf will always print a null-terminating character,
-        // so abort early if only space for one or zero character, as
-        // this would clobber the original string.
-        return false;
-    }
+    ss.grow(end - ss.end());
+    return true;
+}
 
-    // Calculate required length.
-    const int return_code = std::snprintf(nullptr, 0, get_format_code<T>(), value);
-    if (return_code < 0) {
-        return false;
+template<large_int_t Base = 10, integral T>
+bool append_to(small_string_span ss, T value) noexcept {
+    auto [end, err] = std::to_chars(ss.end(), ss.begin() + ss.capacity(), value, Base);
+    if (err != std::errc{}) {
+        // Not enough space, try into a temporary string that *should* be big enough,
+        // and copy whatever we can. 32 characters is enough for all integers and floating
+        // point values encoded on 64 bit or less.
+        small_string<32> fallback;
+        auto [end2, err2] =
+            std::to_chars(fallback.end(), fallback.begin() + fallback.capacity(), value, Base);
+        if (err2 != std::errc{}) {
+            return false;
+        }
+        fallback.grow(end2 - fallback.begin());
+        return append(ss, fallback);
     }
+    ss.grow(end - ss.end());
+    return true;
+}
+#else
+template<floating_point T>
+bool append_to(small_string_span ss, T value) noexcept {
+    return append_constexpr(ss, value);
+}
 
-    // 'return_code' holds the number of characters that are required,
-    // excluding the null-terminating character, which always gets appended,
-    // so we need to +1.
-    const std::size_t length    = static_cast<std::size_t>(return_code) + 1;
-    const bool        could_fit = length <= ss.available();
-
-    const std::size_t offset     = ss.size();
-    const std::size_t prev_space = ss.available();
-    ss.resize(std::min(ss.size() + length, ss.capacity()));
-    std::snprintf(ss.begin() + offset, prev_space, get_format_code<T>(), value);
-
-    // Pop the null-terminating character, always printed unfortunately.
-    ss.pop_back();
-
-    return could_fit;
+template<large_int_t Base = 10, integral T>
+bool append_to(small_string_span ss, T value) noexcept {
+    return append_constexpr(ss, value);
```

Comment:
Isn't this missing here?
```suggestion
    return append_constexpr<Base>(ss, value);
```

Reply:
Was digging into function template argument deduction to make doubly sure that the resolution is unambiguous. Then realized that yes you need it because the caller could set `Base` to be non-default!

I looked to add a test for this but it is an unexposed implementation function. I combed through the other functions that take `Base` and didn't see another instance of this mistake in `snitch_append.cpp`